src/Razor/src/Microsoft.VisualStudio.Editor.Razor/BackgroundParser.cs

```diff
@@ -57,6 +57,7 @@ public ChangeReference QueueChange(SourceChange change, ITextSnapshot snapshot)
         public void Dispose()
         {
             _main.Cancel();
+            _main.Dispose();
```

Comment:
Hmm, this is slightly worrisome. Can you validate that editing a Razor file (without our experimental flag on), closing, re-opening etc. works as expected?

Reply:
The scenario you described works without the experimental flag on when I tried it, could you go into more detail on why you thought that might be a problem scenario?